src/main/java/net/neoforged/neoforge/client/ClientNeoForgeMod.java

```diff
@@ -131,9 +135,16 @@ static void onRegisterModelLoaders(ModelEvent.RegisterLoaders event) {
     }
 
     @SubscribeEvent
-    static void onRegisterReloadListeners(RegisterClientReloadListenersEvent event) {
-        event.registerReloadListener(ObjLoader.INSTANCE);
-        event.registerReloadListener(AnimationLoader.INSTANCE);
+    static void onRegisterReloadListeners(AddClientReloadListenerEvent event) {
+        event.addListener(NeoForgeReloadListeners.CLIENT_MOD_LOADING, ClientModLoader::onResourceReload);
+        event.addListener(NeoForgeReloadListeners.BRANDING, BrandingControl.resourceManagerReloadListener());
+
+        // These run before vanilla reload listeners, so we add them before LANGUAGE, the first vanilla one.
```

Comment:
Is it worth having a special `FIRST` field in `VanillaClientListeners`? And maybe `LAST`?

Reply:
Maybe? The field indices are already ordered by their vanilla order, but sentinel fields that always point to the current first/last values might be useful (or a liability, if we forget to update them...)

Reply:
They're even more of a liability for modders I think.

Reply:
In that we should prefer to not add them? Or in that relying on non-sentinel fields would be dangerous for modders?

Reply:
Yeah I mean that it's a liability for every modder if we don't provide these fields.